File: 2022/july/03-07-2022/wiggle-subsequence.py
class Solution:
    def wiggleMaxLength(self, nums) -> int:
        positive, negative = 1, 1

        if len(nums) < 2:
            return len(nums)

        for i in range(1, len(nums)):
            if nums[i] > nums[i - 1]:
                positive = negative + 1
            elif nums[i] < nums[i - 1]:
                negative = positive + 1

        return max(positive, negative)


print(Solution().wiggleMaxLength([1,7,4,9,2,5]))
print(Solution().wiggleMaxLength([1,17,5,10,13,15,10,5,16,8]))

File: 2022/july/03-07-2022/wiggle-subsequence.rs
impl Solution {
    pub fn wiggle_max_length(nums: Vec<i32>) -> i32 {
        let mut i = 0;
        let mut j = 0;
        let mut ans = 1;
        for k in 1..nums.len() {
            if nums[k] != nums[j] && nums[k].cmp(&nums[j]) != nums[j].cmp(&nums[i]) {
                i = j;
                ans += 1;
            }
            j = k;
        }
        ans
    }
}
